pkg/controllers/cloud_config_sync_controller.go

```diff
@@ -308,3 +338,46 @@ func (r *CloudConfigReconciler) setDegradedCondition(ctx context.Context) error
 	klog.Info("Cloud Config Controller is degraded")
 	return r.syncStatus(ctx, co, conds)
 }
+
+func (r *CloudConfigReconciler) setUpgradeableCondition(ctx context.Context, condition configv1.ConditionStatus, reason string, message string) error {
```

Comment:
nit: in future would be nice to extract this functions to separate file, along with others like set degraded and place it next to each other.

Reply:
i had a similar thought, but i am going to make a jira card about cleaning up these functions we have split across 2 different controller files.